### ISSUE TITLE
Could not locate deployment bucket. Error: The specified bucket does not exist

### ISSUE THREAD

**Issue by djieef** [serverless/serverless#4185]
# This is a *Bug Report*

### What went wrong?

Serverless throws an error when deploying a brand new project to AWS with a custom deployment bucket name. The serverless process exits before even creating the CloudFormation stack. My AWS IAM user account has full administrative privileges. The error message is:

`Could not locate deployment bucket. Error: The specified bucket does not exist`

### What did you expect should have happened?

Serverless should create a new S3 bucket with the name specified in serverless.yml

### What was the config you used?

This is the config snippet I used based on the docs at https://serverless.com/framework/docs/providers/aws/guide/serverless.yml/
```
service: my-service-name
provider:
  name: aws
  runtime: nodejs6.10
  region: ${opt:region, self:custom.defaults.REGION}
  stage: ${opt:stage, self:custom.defaults.SLS_STAGE}
  deploymentBucket:
    name: ${self:service}-${self:provider.stage}-${self:provider.region}-slsbucket
```

### What stacktrace or error message from your provider did you see?

This is the full output of serverless with SLS_DEBUG=*

```
$ sls deploy -s develop -r us-east-1

  Serverless Error ---------------------------------------

  Could not locate deployment bucket. Error: The specified bucket does not exist

  Stack Trace --------------------------------------------

ServerlessError: Could not locate deployment bucket. Error: The specified bucket does not exist
    at AwsPackage.<anonymous> (/usr/local/lib/node_modules/serverless/lib/plugins/aws/package/lib/generateCoreTemplate.js:40:17)
From previous event:
    at AwsPackage.generateCoreTemplate (/usr/local/lib/node_modules/serverless/lib/plugins/aws/package/lib/generateCoreTemplate.js:39:15)
From previous event:
    at Object.AwsPackage.hooks.package:initialize [as hook] (/usr/local/lib/node_modules/serverless/lib/plugins/aws/package/index.js:60:10)
    at /usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:242:55
From previous event:
    at PluginManager.invoke (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:242:22)
    at PluginManager.spawn (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:254:17)
    at Deploy.<anonymous> (/usr/local/lib/node_modules/serverless/lib/plugins/deploy/deploy.js:101:50)
From previous event:
    at Object.Deploy.hooks.before:deploy:deploy [as hook] (/usr/local/lib/node_modules/serverless/lib/plugins/deploy/deploy.js:99:10)
    at /usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:242:55
From previous event:
    at PluginManager.invoke (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:242:22)
    at PluginManager.run (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:261:17)
    at /usr/local/lib/node_modules/serverless/lib/Serverless.js:99:33
    at processImmediate [as _immediateCallback] (timers.js:383:17)
From previous event:
    at Serverless.run (/usr/local/lib/node_modules/serverless/lib/Serverless.js:86:74)
    at /usr/local/lib/node_modules/serverless/bin/serverless:39:50

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Forums:        forum.serverless.com
     Chat:          gitter.im/serverless/serverless

  Your Environment Information -----------------------------
     OS:                     darwin
     Node Version:           4.4.0
     Serverless Version:     1.21.0
```

## Similar or dependent issues:
* #1877 is similar but the suggested workaround of deleting the CloudFormation stack does not apply because the stack is never created.

## Additional Data

* ***Serverless Framework Version you're using***: 1.21.0
* ***Operating System***: MacOS Sierra 10.12.6
* ***Stack Trace***: See above
* ***Provider Error messages***: See above

**Comment by rbhughes** [comment]
## Ditto to above, but different environment

- Serverless Framework Version you're using: 1.21.0
- Operating System: Windows 10 Pro
- Stack Trace: See below
- Provider Error messages: See below

CloudFormation stack does not exists yet.

```
c:\dev\purr-casaubon\serverless>serverless deploy -v --aws-profile sls_admin
Serverless: Packaging service...
Serverless: Excluding development dependencies...

  Serverless Error ---------------------------------------

  The specified bucket does not exist

  Stack Trace --------------------------------------------

ServerlessError: The specified bucket does not exist
    at Response.req.send (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\provider\awsProvider.js:180:20)
    at Request.<anonymous> (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:364:18)
    at Request.callListeners (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:105:20)
    at Request.emit (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:77:10)
    at Request.emit (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:683:14)
    at Request.transition (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:22:10)
    at AcceptorStateMachine.runTo (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\state_machine.js:14:12)
    at C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\state_machine.js:26:10
    at Request.<anonymous> (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:38:9)
    at Request.<anonymous> (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:685:12)
    at Request.callListeners (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:115:18)
    at Request.emit (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:77:10)
    at Request.emit (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:683:14)
    at Request.transition (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:22:10)
    at AcceptorStateMachine.runTo (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\state_machine.js:14:12)
    at C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\state_machine.js:26:10
    at Request.<anonymous> (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:38:9)
    at Request.<anonymous> (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:685:12)
    at Request.callListeners (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:115:18)
    at callNextListener (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:95:12)
    at Request.RESET_RETRY_STATE (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\event_listeners.js:443:9)
    at Request.callListeners (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:101:18)
    at Request.emit (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:77:10)
    at Request.emit (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:683:14)
    at Request.transition (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:22:10)
    at AcceptorStateMachine.runTo (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\state_machine.js:14:12)
    at C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\state_machine.js:26:10
    at Request.<anonymous> (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:38:9)
    at Request.<anonymous> (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:685:12)
    at Request.callListeners (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:115:18)
    at Request.emit (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:77:10)
    at Request.emit (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:683:14)
    at Request.transition (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:22:10)
    at AcceptorStateMachine.runTo (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\state_machine.js:14:12)
    at C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\state_machine.js:26:10
    at Request.<anonymous> (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:38:9)
    at Request.<anonymous> (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:685:12)
    at Request.callListeners (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:115:18)
    at callNextListener (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:95:12)
    at Request.requestBucketRegion (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\services\s3.js:601:14)
    at Request.callListeners (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:101:18)
    at Request.emit (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:77:10)
    at Request.emit (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:683:14)
    at Request.transition (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:22:10)
    at AcceptorStateMachine.runTo (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\state_machine.js:14:12)
    at C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\state_machine.js:26:10
    at Request.<anonymous> (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:38:9)
    at Request.<anonymous> (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:685:12)
    at Request.callListeners (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:115:18)
    at Request.emit (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:77:10)
    at Request.emit (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:683:14)
    at Request.transition (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:22:10)
    at AcceptorStateMachine.runTo (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\state_machine.js:14:12)
    at C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\state_machine.js:26:10
    at Request.<anonymous> (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:38:9)
    at Request.<anonymous> (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\request.js:685:12)
    at Request.callListeners (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:115:18)
    at callNextListener (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\sequential_executor.js:95:12)
    at IncomingMessage.onEnd (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\node_modules\aws-sdk\lib\event_listeners.js:269:13)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:188:7)
    at endReadableNT (_stream_readable.js:975:12)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickDomainCallback (internal/process/next_tick.js:128:9)
From previous event:
    at persistentRequest (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\provider\awsProvider.js:167:14)
    at doCall (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\provider\awsProvider.js:146:9)
    at BbPromise (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\provider\awsProvider.js:157:14)
From previous event:
    at persistentRequest (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\provider\awsProvider.js:144:38)
    at AwsProvider.request (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\provider\awsProvider.js:160:12)
    at AwsDeploy.getMostRecentObjects (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\deploy\lib\checkForChanges.js:33:26)
From previous event:
    at AwsDeploy.checkForChanges (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\deploy\lib\checkForChanges.js:20:8)
From previous event:
    at Object.aws:deploy:deploy:checkForChanges [as hook] (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\deploy\index.js:102:10)
    at BbPromise.reduce (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\classes\PluginManager.js:242:55)
From previous event:
    at PluginManager.invoke (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\classes\PluginManager.js:242:22)
    at PluginManager.spawn (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\classes\PluginManager.js:254:17)
    at AwsDeploy.BbPromise.bind.then (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\deploy\index.js:91:48)
From previous event:
    at Object.deploy:deploy [as hook] (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\deploy\index.js:87:10)
    at BbPromise.reduce (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\classes\PluginManager.js:242:55)
From previous event:
    at PluginManager.invoke (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\classes\PluginManager.js:242:22)
    at PluginManager.run (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\classes\PluginManager.js:261:17)
    at variables.populateService.then (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\Serverless.js:99:33)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
From previous event:
    at Serverless.run (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\lib\Serverless.js:86:74)
    at serverless.init.then (C:\Users\r-b-h\AppData\Roaming\npm\node_modules\serverless\bin\serverless:39:50)

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Forums:        forum.serverless.com
     Chat:          gitter.im/serverless/serverless

  Your Environment Information -----------------------------
     OS:                     win32
     Node Version:           7.8.0
     Serverless Version:     1.21.0

```

**Comment by rbhughes** [comment]
Did a bit more testing. Neither purging contents of `.serverless` directory nor testing four prior versions of serverless package worked. However, changing only the `service` name in `serverless.yml` allows a normal deployment/removal. Did a prior bucket delete fail and the "unique" name get stuck? Don't see it in AWS console.

**Comment by chasebleyl** [comment]
Has anyone found a workaround or a fix for changing the `deploymentBucket` attribute? I am running into issues where my default deployment bucket name is too long, which makes setting up IAM policies a little tricky.

**Comment by daves0** [comment]
Ditto to above, but different environment

    Serverless Framework Version you're using: 1.20.2
    Operating System: Ubuntu 16.10
    Stack Trace: Same as the initial report
    Provider Error messages: Same as the initial report

Again like the initial report, I'm using a custom bucket name and the CloudFormation stack does not exist yet.  Also perhaps of not, there is NO .serverless directory created, as there in other successful deploys I've done for other projects.  This appears to be stopping very early in the process.

**Comment by daves0** [comment]
I found a work-around.  I created the bucket that was missing per the error message, and then ran

sls remove

... and the remove process ran without error.  I then was able to run

sls deploy

without issue.  Yay!  Hope this helps track down the bug in the sls deploy process.

**Comment by rbhughes** [comment]
TLDR: wait a very long time, check CloudFormation console, delete failed serverless stack.
After ignoring this issue for weeks, I checked CloudFormation and (finally) the failed stack showed up (there were no stacks at all). I was able to delete it from the AWS console without incident and could deploy the original, failing serverless.yml without problem. I have no idea how long the interval was between when I first saw the "bucket does not exist" error and the missing stack's appearance in the CloudFormation console, but it was over a week.

**Comment by Vadorequest** [comment]
Similar issue, when I define the `deploymentBucket.name` then it fails to deploy. I wanted to use a custom name to have a better-looking url and makes it easier to reason with. I tried to create the S3 bucket by hand but still the same issue.

```
(studylink-dev) ➜  simulator-feedback git:(master) ✗ serverless deploy -v
Serverless: Load command run
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command create
Serverless: Load command install
Serverless: Load command package
Serverless: Load command deploy
Serverless: Load command deploy:function
Serverless: Load command deploy:list
Serverless: Load command deploy:list:functions
Serverless: Load command invoke
Serverless: Load command invoke:local
Serverless: Load command info
Serverless: Load command logs
Serverless: Load command login
Serverless: Load command logout
Serverless: Load command metrics
Serverless: Load command print
Serverless: Load command remove
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command slstats
Serverless: Load command plugin
Serverless: Load command plugin
Serverless: Load command plugin:install
Serverless: Load command plugin
Serverless: Load command plugin:uninstall
Serverless: Load command plugin
Serverless: Load command plugin:list
Serverless: Load command plugin
Serverless: Load command plugin:search
Serverless: Load command emit
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command webpack
Serverless: Invoke deploy
Serverless: Invoke package
Serverless: Invoke aws:common:validate
Serverless: Invoke aws:common:cleanupTempDir
Serverless: Invoke webpack:validate
Serverless: Invoke webpack:compile
Serverless: Bundling with Webpack...
ts-loader: Using typescript@2.5.2 and /Users/vadorequest/dev/student-loan-simulator-serverless/aws/simulator-feedback/tsconfig.json
Time: 7451ms
          Asset     Size  Chunks                    Chunk Names
    feedback.js  3.05 MB       0  [emitted]  [big]  feedback
feedback.js.map  4.23 MB       0  [emitted]         feedback
   [0] ./node_modules/aws-sdk/lib/core.js 2.2 kB {0} [built]
   [1] ./node_modules/aws-sdk/lib/node_loader.js 3.08 kB {0} [built]
  [15] ./node_modules/aws-sdk/lib/protocol/rest.js 4.39 kB {0} [built]
  [24] external "crypto" 42 bytes {0} [not cacheable]
  [88] ./feedback.ts 5.16 kB {0} [built]
  [89] ./node_modules/uuid/v1.js 3.34 kB {0} [built]
  [90] ./node_modules/uuid/lib/rng.js 246 bytes {0} [built]
  [91] ./node_modules/uuid/lib/bytesToUuid.js 699 bytes {0} [built]
  [92] ./db/dynamodb.ts 227 bytes {0} [built]
  [93] ./node_modules/aws-sdk/lib/aws.js 134 bytes {0} [built]
 [226] external "querystring" 42 bytes {0} [not cacheable]
 [255] ./node_modules/aws-sdk/clients/all.js 5.24 kB {0} [built]
 [352] ./node_modules/aws-sdk/clients/directoryservice.js 604 bytes {0} [built]
 [700] ./node_modules/aws-sdk/clients/transcribeservice.js 625 bytes {0} [built]
 [703] ./utils/response.ts 486 bytes {0} [built]
    + 689 hidden modules
Serverless: Invoke webpack:package
Serverless: Zip service: /Users/vadorequest/dev/student-loan-simulator-serverless/aws/simulator-feedback/.webpack/service [109 ms]
Serverless: Packaging service...
Serverless: Remove /Users/vadorequest/dev/student-loan-simulator-serverless/aws/simulator-feedback/.webpack
Serverless: Invoke aws:package:finalize
Serverless: Invoke aws:common:moveArtifactsToPackage
Serverless: Invoke aws:common:validate
 
  Serverless Error ---------------------------------------
 
  Could not locate deployment bucket. Error: The specified bucket does not exist
 
  Stack Trace --------------------------------------------
 
ServerlessError: Could not locate deployment bucket. Error: The specified bucket does not exist
    at BbPromise.resolve.then.then.catch.err (/Users/vadorequest/.nvm/versions/node/v8.9.4/lib/node_modules/serverless/lib/plugins/aws/deploy/lib/existsDeploymentBucket.js:26:15)
From previous event:
    at AwsDeploy.existsDeploymentBucket (/Users/vadorequest/.nvm/versions/node/v8.9.4/lib/node_modules/serverless/lib/plugins/aws/deploy/lib/existsDeploymentBucket.js:25:13)
    at AwsDeploy.BbPromise.bind.then.then (/Users/vadorequest/.nvm/versions/node/v8.9.4/lib/node_modules/serverless/lib/plugins/aws/deploy/index.js:81:25)
From previous event:
    at Object.before:deploy:deploy [as hook] (/Users/vadorequest/.nvm/versions/node/v8.9.4/lib/node_modules/serverless/lib/plugins/aws/deploy/index.js:78:10)
    at BbPromise.reduce (/Users/vadorequest/.nvm/versions/node/v8.9.4/lib/node_modules/serverless/lib/classes/PluginManager.js:372:55)
From previous event:
    at PluginManager.invoke (/Users/vadorequest/.nvm/versions/node/v8.9.4/lib/node_modules/serverless/lib/classes/PluginManager.js:372:22)
    at PluginManager.run (/Users/vadorequest/.nvm/versions/node/v8.9.4/lib/node_modules/serverless/lib/classes/PluginManager.js:403:17)
    at variables.populateService.then (/Users/vadorequest/.nvm/versions/node/v8.9.4/lib/node_modules/serverless/lib/Serverless.js:102:33)
    at runCallback (timers.js:789:20)
    at tryOnImmediate (timers.js:751:5)
    at processImmediate [as _immediateCallback] (timers.js:722:5)
From previous event:
    at Serverless.run (/Users/vadorequest/.nvm/versions/node/v8.9.4/lib/node_modules/serverless/lib/Serverless.js:89:74)
    at serverless.init.then (/Users/vadorequest/.nvm/versions/node/v8.9.4/lib/node_modules/serverless/bin/serverless:42:50)
    at <anonymous>
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Forums:        forum.serverless.com
     Chat:          gitter.im/serverless/serverless
 
  Your Environment Information -----------------------------
     OS:                     darwin
     Node Version:           8.9.4
     Serverless Version:     1.26.0

```

serverless.yml
```
service:
  name: simulator-feedback

plugins:
  - serverless-webpack

provider:
  name: aws
  runtime: nodejs6.10
  versionFunctions: false # See https://serverless.com/framework/docs/providers/aws/guide/functions#versioning-deployed-functions
  stage: development
  region: eu-west-3
  deploymentBucket:
    name: sl-simulator

  # 'iamRoleStatement' defines the permission policy for the Lambda function.
  # In this case Lambda functions are granted with permissions to access DynamoDB.
  iamRoleStatements:
    - Effect: Allow
      Action:
        - dynamodb:DescribeTable
        - dynamodb:Query
        - dynamodb:Scan
        - dynamodb:GetItem
        - dynamodb:PutItem
        - dynamodb:UpdateItem
#        - dynamodb:DeleteItem
      Resource: "arn:aws:dynamodb:eu-west-3:035907498810:table/simulator-feedback"

functions:
  create:
    handler: feedback.create
    events:
      - http:
          method: post
          path: feedback/create # path: url path is /create
          cors: true
  update:
    handler: feedback.update
    events:
      - http:
          method: put
          path: feedback/update # path: url path is /update
          cors: true

```

**Comment by eahefnawy** [comment]
As mentioned in #5042 ... the custom deployment bucket is meant for providing existing bucket names. Creating those **custom** buckets is beyond the scope of the framework. If you create the bucket either manually, or any other provisioning tool, it should work out.

Thanks all for the discussion.

Closing...

**Comment by stefanomiccoli** [comment]
is there any technical good reason that i dont get to avoid the creation of the deployment bucket with custom naming? 
at the end of the day, serverless framework is able to generate them, but the naming is sometimes really not desirable. However forcing to create them manually or via other provisioning tools add unnecessary complexity to carry aws.region from one tool to the other to ensure that the bucket is created in the same region of the lambda...

**Comment by cschroed-usgs** [comment]
There is a handy plugin that creates custom serverless deployment S3 buckets for you.
https://www.serverless.com/plugins/serverless-deployment-bucket
